### PR TITLE
feat: remove through2 dependency

### PR DESCRIPTION
**Pull request by laljikanjareeya**
Towards #708

### CODE REVIEW

File: src/table.ts [hunk 1]
```diff
@@ -810,24 +809,21 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
...
-      rowStream = pumpify.obj([
-        requestStream,
-        chunkTransformer,
-        through.obj((rowData, enc, next) => {
-          if (
-            chunkTransformer._destroyed ||
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            (userStream as any)._writableState.ended
-          ) {
+      const transform = new Transform({
```
Comment (stephenplusplus) [inline src/table.ts:812]: Can we call this "'toRowStream" or something similar to indicate what it's doing?

File: src/table.ts [hunk 2]
```diff
@@ -1561,19 +1557,24 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
       appProfileId: this.bigtable.appProfileId,
     };
 
+    const transform = new Transform({
```
Comment (stephenplusplus) [inline src/table.ts:1560]: Is there a more descriptive name we can use for this stream as well?

File: system-test/read-rows-acceptance-tests.ts [hunk 3]
```diff
@@ -69,31 +71,29 @@ describe('Read Row Acceptance tests', () => {
...
+            );
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            let readRowsResponse: any = {chunks: [cellChunk]};
+            readRowsResponse = ReadRowsResponse.create(readRowsResponse);
+            readRowsResponse = ReadRowsResponse.toObject(readRowsResponse, {
+              defaults: true,
+              longs: String,
+              oneofs: true,
+            });
+            stream.emit('data', readRowsResponse);
```
Comment (stephenplusplus) [inline system-test/read-rows-acceptance-tests.ts:91]: `stream.push()` won't work anymore?

**Review by stephenplusplus — CHANGES_REQUESTED** [review]

File: src/table.ts (hunk 1 above)
Comment (laljikanjareeya) [reply src/table.ts]: Done!

File: system-test/read-rows-acceptance-tests.ts (hunk 3 above)
Comment (laljikanjareeya) [reply system-test/read-rows-acceptance-tests.ts]: yes that works fine.

**Comment by laljikanjareeya** [comment]
@stephenplusplus Thanks for the review.

Two tests were failing:
1. invalid - duplicate row key
2. invalid - new row missing row key

I have also cross verify it with `through2` version `4.0.2` these tests are also failing as they have removed `destroy()`.
Based on that I just change the test cases.

**Review by stephenplusplus — APPROVED** [review]

**Pull request merged**